routes/api.php

```diff
@@ -13,6 +13,45 @@
 | is assigned the "api" middleware group. Enjoy building your API!
 |
 */
+Route::post('/filterRooms', function (Request $request) {
+
+    // None of the request fields are mandatory, only
+    // filter the ones provided from request
+    $validation = $request->validate([
+        'capacity_standing' => 'numeric|min:0|not_in:0',
+        'capacity_sitting' => 'numeric|min:0|not_in:0',
+        'food' => ['boolean'],
+        'alcohol' => ['boolean'],
+        'a_v_permitted' => ['boolean'],
+        'projector' => ['boolean'],
+        'television' => ['boolean'],
+        'computer' => ['boolean'],
+        'whiteboard' => ['boolean'],
+        'sofas' => 'numeric|min:0|not_in:0',
+        'coffee_tables' => 'numeric|min:0|not_in:0',
+        'tables' => 'numeric|min:0|not_in:0',
+        'chairs' => 'numeric|min:0|not_in:0',
+        'ambiant_music' => ['boolean'],
+        'sale_for_profit' => ['boolean'],
+        'fundraiser' => ['boolean'],
+    ]);
+
+    // If param value boolean, filter for boolean and if
+    // numeric filter greater than integer in that column
+    $query = Room::query();
+    foreach ($request->toArray() as $key => $value){
+        if(is_numeric($value)){
+          $query->where('attributes->' . $key, '>=', $value);
+        }else{
+          if(is_bool($value)) {
+            $query->where('attributes->' . $key, $value);
+          }
+        }
+
+    }
+    return response()->json($query->get());
+
```

Comment:
This absolutely needs to be put into a controller. This is too bulky to be left in the routes file.

Reply:
Do you think I should add this as a method in the rooms controller as opposed to making a whole new one?

Reply:
I would make a new controller. Just to keep it compartmentalized